code/modules/scavstation_shared_content/items/egg_yinglet.dm

```diff
@@ -0,0 +1,86 @@
+//yinglet nests and yinglet eggs
+
+//nest
+/obj/structure/yinglet_nest
+	name = "eggbed"
+	icon = 'code/modules/scavstation_shared_content/icons/egg_yinglet.dmi'
+	icon_state = "base"
+	w_class = ITEM_SIZE_HUGE
+	var/has_egg = TRUE
+
+/obj/structure/yinglet_nest/empty
+	has_egg = FALSE
+
+/obj/structure/yinglet_nest/Initialize()
+	. = ..()
+	if(has_egg)
+		new /obj/item/yinglet_egg(src)
+	update_icon()
+	update_desc()
+
+/obj/structure/yinglet_nest/proc/update_desc()
+	desc = "A pile of plain but very soft fabric arranged to carefully hold an egg, as well as prevent the newly hatched younglet from smashing their face when they flop out."
+	if(has_egg && contents.len)
+		for(var/obj/item/I in contents)
+			desc += " There's \a [I] sitting in it."
+
+/obj/structure/yinglet_nest/attack_hand(mob/user)
+	. = ..()
+	if(has_egg && user.a_intent == I_HELP)
+		for(var/obj/item/I in contents)
```

Comment:
If there's more than one object in contents, this will show several messages and knock their cooldown each time. Maybe...
````
if(has_egg && user.a_intent == I_HELP && contents.len)
	var/obj/item/I = pick(contents)
	src.visible_message(SPAN_NOTICE("\The [user] pets \the [I]. Grow up big and strong now, okay?"))
	user.setClickCooldown(1.5 SECONDS)
	return
````
?

Reply:
TBH there should be no circumstance where there is more than one object.

To tell the truth I don't think for loops is ideal for what I want. I was only using them because i want to avoid list fuckery. What would you recommend

Reply:
Tbh the only reason I used for loops was i didn't know a better way to do it.

There should never be more than one item in a nest at any given time really. Pick works as well i guess but if we knew there was only one item what would we do?

Reply:
Either `contents[1]` or `locate(/obj/item) in contents`.

Reply:
locate()?

Reply:
This isn't actually resolved, that I can see?